webapp/src/components/admin_console_settings/recordings/enable_recordings.tsx

```diff
@@ -31,25 +30,34 @@ import {
     Title,
     Text,
     Footer,
-    FooterText,
+    FooterText, leftCol, rightCol,
 } from 'src/components/admin_console_settings/common';
 
 const EnableRecordings = (props: CustomComponentProps) => {
     const dispatch = useDispatch();
     const restricted = useSelector(isOnPremNotEnterprise);
     const cloud = useSelector(isCloud);
     const stats = useSelector(adminStats);
-
-    const leftCol = 'col-sm-4';
-    const rightCol = 'col-sm-8';
-
-    // Webapp doesn't pass the placeholder setting.
-    const placeholder = manifest.settings_schema?.settings.find((e) => e.key === 'EnableRecordings')?.placeholder || '';
+    const [enabled, setEnabled] = useState(() => props.value === 'true');
 
     const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
         props.onChange(props.id, e.target.value === 'true');
+        setEnabled(e.target.value === 'true');
     };
 
+    const onSave = useCallback(async () => {
+        dispatch(setRecordingsEnabled(enabled));
+        return {};
+    }, [dispatch, enabled]);
+
+    useEffect(() => {
+        props.registerSaveAction(onSave);
+
+        return () => {
+            props.unRegisterSaveAction(onSave);
+        };
+    }, [onSave, props]);
```

Comment:
This is interesting because intuitively I'd expect the settings to immediately show upon selecting "true". The reason for that is that as an admin I may want to configure things properly before actually enabling the feature. So from a UX perspective I think it may actually be better to retain that behaviour if possible. Thoughts?

Reply:
Looks like it just took a little tweaking to get the correct behaviour. Let me know what you think.